python/paddle/fluid/tests/unittests/dygraph_to_static/test_word2vec.py

```diff
@@ -0,0 +1,325 @@
+
+
+word2id_freq, word2id_dict, id2word_dict = build_dict(corpus)
+vocab_size = len(word2id_freq)
+print("there are totoally %d different words in the corpus" % vocab_size)
+for _, (word, word_id) in zip(range(50), word2id_dict.items()):
+    print("word %s, its id %d, its word freq %d" %
+          (word, word_id, word2id_freq[word_id]))
+
+
+def convert_corpus_to_id(corpus, word2id_dict):
+    new_corpus = []
+    for line in corpus:
+        new_line = [
+            word2id_dict[word]
+            if word in word2id_dict else word2id_dict['[oov]'] for word in line
+        ]
+        new_corpus.append(new_line)
+    return new_corpus
+
+
+corpus = convert_corpus_to_id(corpus, word2id_dict)
+
+
+def subsampling(corpus, word2id_freq):
+    def keep(word_id):
+        return random.uniform(0, 1) < math.sqrt(1e-4 / word2id_freq[word_id] *
+                                                len(corpus))
+
+    new_corpus = []
+    for line in corpus:
+        new_line = [word for word in line if keep(word)]
+        new_corpus.append(line)
+    return new_corpus
+
+
+corpus = subsampling(corpus, word2id_freq)
+
+
+def build_data(corpus,
+               word2id_dict,
+               word2id_freq,
+               max_window_size=3,
+               negative_sample_num=10):
+
+    dataset = []
+
+    for line in corpus:
+        for center_word_idx in range(len(line)):
+            window_size = random.randint(1, max_window_size)
+            center_word = line[center_word_idx]
+
+            positive_word_range = (max(0, center_word_idx - window_size), min(
+                len(line) - 1, center_word_idx + window_size))
+            positive_word_candidates = [
+                line[idx]
+                for idx in range(positive_word_range[0], positive_word_range[1]
+                                 + 1)
+                if idx != center_word_idx and line[idx] != line[center_word_idx]
+            ]
+
+            if not positive_word_candidates:
+                continue
+
+            for positive_word in positive_word_candidates:
+                dataset.append((center_word, positive_word, 1))
+
+            i = 0
+            while i < negative_sample_num:
+                negative_word_candidate = random.randint(0, vocab_size - 1)
+
+                if negative_word_candidate not in positive_word_candidates:
+                    dataset.append((center_word, negative_word_candidate, 0))
+                    i += 1
+
+    return dataset
+
+
+dataset = build_data(corpus, word2id_dict, word2id_freq)
+for _, (center_word, target_word, label) in zip(range(50), dataset):
+    print("center_word %s, target %s, label %d" %
+          (id2word_dict[center_word], id2word_dict[target_word], label))
+
+
+def build_batch(dataset, batch_size, epoch_num):
+
+    center_word_batch = []
+    target_word_batch = []
+    label_batch = []
+    eval_word_batch = []
+
+    for epoch in range(epoch_num):
+
+        #random.shuffle(dataset)
```

Comment:
Is this commented code necessary?

Reply:
No, the commented code is in original code. I commented because the random.shuffle can cause the input to dygraph/static be different thus our check can fail.